app/models/post.rb

```diff
@@ -1,2 +1,20 @@
 class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments, foreign_key: 'posts_id'
+  has_many :likes, foreign_key: 'posts_id'
+
+  def self.post_counter(user_id)
+    user = User.find(user_id)
+    if user.posts_count.nil?
+      user.posts_count = 1
+    else
+      user.posts_count += 1
+    end
+
+    user.save
+  end
```

Comment:
- [ ] [OPTIONAL] You're doing a great job, but this is the long way to do this. Since there's a belong-to relationship between a post and a user, you can easily do this like this 👇
```
def post_counter
  author.increment!(:posts_counter)
end
```

You can then use the `:after_save` method which is one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) to call the `posts_counter` method every time a new post has been saved to the database. After that, you may now make the `posts_counter` method private.